src/db/car/forest.rs

```diff
@@ -268,24 +274,22 @@ impl Encoder {
         offset += header_len;
 
         // Write seekable zstd and collect a mapping of CIDs to frame_offset+data_offset.
-        let mut cid_mapping = Vec::new();
+        let mut cid_to_frame_offset = CidHashMap::new();
         while let Some((cids, zstd_frame)) = stream.try_next().await? {
-            for cid in cids {
-                cid_mapping.push((Hash::from(cid), offset as FrameOffset));
-            }
+            cid_to_frame_offset.extend(cids.into_iter().map(|cid| (cid, offset as u64)));
             sink.write_all(&zstd_frame).await?;
-            offset += zstd_frame.len();
+            offset += zstd_frame.len()
         }
 
         // Create index
-        let index_offset = offset as u64 + 8;
-        let builder = CarIndexBuilder::new(cid_mapping.into_iter());
-        write_skip_frame_header_async(sink, builder.encoded_len()).await?;
-        builder.write_async(sink).await?;
+        let mut index = vec![];
+        index::write(cid_to_frame_offset, &mut index).expect("Vec has infallible IO");
+        write_skip_frame_header_async(sink, index.len().try_into().unwrap()).await?;
+        sink.write_all(&index).await?;
```

Comment:
Streaming the index to disk would save us several GiBs of memory. My gut feeling is that this optimization belongs in this PR.

Reply:
I've used `SyncIoBridge`, but that requires `#[tokio::test(flavor = "multi_thread")]`, which feels a bit ugly